jax/_src/numpy/polynomial.py

```diff
@@ -292,6 +295,25 @@ def polymul(a1, a2, *, trim_leading_zeros=False):
   val = convolve(a1, a2, mode='full')
   return val
 
+@_wraps(np.polydiv, lax_description=_LEADING_ZEROS_DOC)
+def polydiv(u, v, *, trim_leading_zeros=False):
+  _check_arraylike("polydiv", u, v)
+  u, v = _promote_dtypes_inexact(u, v)
+  w = u[0] + v[0] # calculate type
+  m = len(u) - 1
+  n = len(v) - 1
+  scale = 1. / v[0]
+  q = zeros(max(m - n + 1, 1), dtype = w.dtype) # force same dtype
+  r = u.astype(w.dtype)
+  for k in range(0, m-n+1):
+    d = scale * r[k]
+    q = q.at[k].set(d)
+    r = r.at[k:k+n+1].add(-d*v)
+  if trim_leading_zeros:
+    return q, trim_zeros(
+      where((~(abs(r)<finfo(w.dtype).eps)).cumsum().astype(bool), r, 0), trim='f')
```

Comment:
Numpy's version of this is here: https://github.com/numpy/numpy/blob/d410c8da4b9d7ef7e7be355f6804bc0f2ee9a2d5/numpy/lib/polynomial.py#L1042-L1043

Interestingly, the `rtol=1E-14` has no effect since it's multiplied by the second argument, which is zero. Since `atol` defaults to `1E-8`, this means it's effectively equivalent to using `1E-8` in place of `finfo(w.dtype).eps`, suggesting thta 1E-8 is the correct choice for `float64`.

It suggests that using `eps` directly as the cutoff is too tight a constraint. So maybe `np.sqrt(finfo(w.dtype).eps)` would be closer to the spirit of what numpy does here?

Reply:
Also, I think the implementation here is slightly off, because this will zero-out *every* small entry in the result. The elements that are not trimmed should be left unmodified. It may be better to define an alternate `trim_zeros` method that accounts for the tolernace.

Reply:
I changed the code, so the zero-out where stop when it found the first big element. ```trim_zeros``` does the same, it stops when it found the first non-zero element. In this case, both zero-out and trim_zeros visit from left to right. When they stop, rest of the elements are left unmodified.

Reply:
Looks like the new changes didn't make it into the push?

Reply:
It's the ```cumsum```,
A mask like ```[False, False, True, False, True]``` becomes ```[False, False, True, True, True]```.

Reply:
Oh, I see, thanks. I also had a comment about the eps needing to be adjusted as well – what do you think?

Reply:
Yeah, I added the ```sqrt``` in the latest code. Sorry, I squashed all the commits so you didn't notice.